### PR TITLE
feat: add CosmosDB read-only query endpoint to the admin API

### PR DESCRIPTION
**Pull request by bennerv**
## Summary
- Adds `POST /admin/v1/cosmos/query` endpoint to the admin API for direct CosmosDB read access in production environments
- Validates queries against mutating keywords (UPDATE, DELETE, INSERT, CREATE, etc.) with string-literal-aware word-boundary matching
- Accepts container name, SQL query, optional partition key, and max items limit; returns raw JSON documents
- MISE-gated via the existing Istio AuthorizationPolicy — no additional config needed

## Test plan
- [x] Unit tests for keyword validation (35 cases covering edge cases: string literals, field name substrings, case insensitivity)
- [x] E2E tests verified locally against dev environment:
  - Valid SELECT query returns results from Resources container
  - Mutating keywords are rejected with 400
  - Partition key scoping works
  - Empty result sets return correctly
- [ ] CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by bennerv** [comment]
This PR was mostly AI-authored. I'll take it out of draft once I've fully reviewed it.

File: admin/server/handlers/cosmosquery/query.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+const (
+	defaultMaxItems = 100
+	maxMaxItems     = 1000
+	maxRequestBody  = 1 * 1024 * 1024 // 1MB
+)
+
+var mutatingKeywordPattern *regexp.Regexp
+
+func init() {
+	keywords := []string{
```
Comment (bennerv) [inline admin/server/handlers/cosmosquery/query.go:41]: todo: figure out if the query client we use actually allows mutating records.  if it doesn't no need to filter out key words here.

File: admin/server/handlers/cosmosquery/query.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	if results == nil {
+		results = []json.RawMessage{}
+	}
+
+	resp := QueryResponse{
+		Results: results,
+		Count:   len(results),
+	}
+
+	_, err = arm.WriteJSONResponse(w, http.StatusOK, resp)
```
Comment (bennerv) [inline admin/server/handlers/cosmosquery/query.go:190]: todo: we should stream the result instead of limiting max items so we can see the full response from the client side?

**Comment by bennerv** [comment]
/close

will be implemented as part of: https://redhat.atlassian.net/browse/ARO-29536
